lib/travis_event/listener.rb

```diff
@@ -0,0 +1,52 @@
+module TravisEvent
+  module Handlers
+    def self.handlers_for
+      Dir.glob("lib/travis_event/handlers/*.rb").collect do |f|
+        path = Pathname.new(f).relative_path_from(Pathname.new("lib")).to_s
+        path.chomp(".rb").classify.constantize
+      end
+    end
+  end
+
+  class Listener
+    ALL_EVENTS = %w(build:created build:started build:finished job:created job:started job:finished)
+
+    # repos: [Travis::Repository.find("rails/rails")]
+    def self.monitor(*repos)
+      Travis.listen(*repos) do |stream|
+        stream.on(*ALL_EVENTS) do |event|
+          args = extract_args(event)
+          if args
+            Handlers.handlers_for.each do |handler|
```

Comment:
You should cache the handlers in the class method like we did with the commit_monitor, or this will get expensive.